index.js

```diff
@@ -149,6 +150,90 @@ class GithubScm extends Scm {
     }
 
     /**
+     * Adds the Screwdriver webhook to the Github repository
+     * @method _addWebhook
+     * @param  {Object}    config            Config object
+     * @param  {String}    config.scmUri     The SCM URI to add the webhook to
+     * @param  {String}    config.token      Service token to authenticate with Github
+     * @param  {String}    config.webhookUrl The URL to use for the webhook notifications
+     * @return {Promise}                     Resolve means operation completed without failure.
+     */
+    _addWebhook(config) {
+        const searchForHook = (scmInfo, url, page) => {
+            const getHooks = this.breaker.runCommand({
+                action: 'getHooks',
+                token: config.token,
+                params: {
+                    owner: scmInfo.owner,
+                    repo: scmInfo.repo,
+                    page,
+                    per_page: WEBHOOK_PAGE_SIZE
+                }
+            });
+            const filterForHook = getHooks.then((hooks) => {
+                let found = null;
+
+                hooks.some((hook) => {
+                    const currentHook = hoek.reach(hook, 'config.url');
+
+                    if (currentHook === url) {
+                        found = hook;
+                    }
+
+                    return found;
+                });
+
+                return found;
+            });
+
+            return Promise.all([
+                getHooks,
+                filterForHook
```

Comment:
Right. `_addWebhook` is fairly complex, which is why there are attempts to reduce cyclomatic complexity when it can. Finding the Screwdriver hook in the list of webhooks is what contributes the most to this, and doing it in the Bitbucket module hasn't been fun either 😅 

I've always been open to changing it. Updating webhooks requires a lot of different information and resources tied to the object, so we can't put it outside like we do with `getInfo`.

I don't like having the `findInHook` passing values forward in the chain, since it presents many of the same complaints we had about `async.waterfall`